### DISCUSSION TITLE
Designer XAML Exception - No trace

### DISCUSSION THREAD

**Discussion by markilg** [AvaloniaUI/Avalonia#9322] [Q&A]
Receive an invalid markup randomly in designer for control and/or window.  

"Unable to load requested types" in the Error Dialog with the following Debug Error: 

System.Xaml.XamlException: Unable to load one or more of the requested types.
Could not load type 'IndexerClosure_866593db90bb4f9280a91671a23e424e' from assembly 'a3a03ba952344b5fbd0bb1aae3034105, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
11:48:48.491 [Verbose] 56148 Finished PreviewerProcess.OnMessageAsync()

Problem typically arises when using a data template but not sure if this is the culprit.  After the error occurs, removing all content from the xaml will still produce the error on a blank control.  Cannot trace any further on the failed type as there are no indication in the XAML there is an error.  Program compiles fine and runs fine.

**Comment by timunie** [comment]
Without seeing your code  hard to tell what's wrong. Can you upload your project to GitHub and post the link here?

But I doubt it's a bug of Avalonia. Can I convert this to a Q&A for now?

**Comment by markilg** [comment]
Unfortunately I cannot post the project to GitHub due to firewall constraints but you can push to Q&A.  I'm thinking it may be a caching issue.  What default path does avalonia use for the designer?

**Reply by timunie** [reply]
I think the designer is using the bin dir. but it's quite limited for actions which needs runtime support. You can probably add a check like this: 

```cs
if (Design.IsDesignMode)
{
    ....
}
else 
{
    do your caching here
}
```

See: http://reference.avaloniaui.net/api/Avalonia.Controls/Design/E9AECCD6

Other than that without additional code, help is also limited. If you can't share your entire project, try to strip your issue down to a minimal repo which fails. Then we can have a deeper look. 

Happy coding
Tim
